### PR TITLE
Jest fix for uncaught promise

### LINKED ISSUE #24: Fails with jest 27
Upgrading to jest 27 causes an Uncaught promise: Error: aborted

This branch has a "fix" but it is unclear if it is still testing the same behavior

https://github.com/GMOD/abortable-promise-cache/tree/fails_jest27

Specific test
```
yarn test -t 'cache 2 requests, both aborted, one pre-aborted, and fill aborted'
```

### PR DESCRIPTION
**Pull request by cmdcolin**
This appears to fix the uncaught promise found in #24

Fixes #24 

Also changes the promise chaining on newEntry.promise to actually store the modified then/catch on newEntry.promise instead of those being "separate chains" (.then and .catch return a new promise) but this does not affect the test results

### CODE REVIEW

File: src/AbortablePromiseCache.ts [hunk 1]
```diff
@@ -177,6 +169,13 @@ export default class AbortablePromiseCache<T, U> {
         'second get argument appears to be an AbortSignal, perhaps you meant to pass `null` for the fill data?',
       )
     }
+
+    //check pre-aborted
+    if (signal?.aborted) {
```
Comment (cmdcolin) [inline src/AbortablePromiseCache.ts:174]: main contribution to fix issue here

**Comment by cmdcolin** [comment]
@rbuels i saw you merged some stuff, any interest in this PR+release new version?

**Comment by cmdcolin** [comment]
note also that dependabot stuff is not that impactful on library codebases since e.g. when the library is installed the yarn.locks are ignored

**Comment by rbuels** [comment]
Yeah I'm working on that PR now

**Comment by rbuels** [comment]
closed in favor of #26

**Pull request closed without merging**